packages/mco/components/create-dr-policy/selected-cluster-validator.tsx

```diff
@@ -239,123 +243,263 @@ const PeeringValidationMessage: React.FC<PeeringValidationMessageProps> = ({
+  const mapClusterNameToSCDetails: Map<
+    ClusterNameType,
+    StorageClassDetailsType[]
+  > = new Map();
+  if (searchLoaded && searchError == null) {
+    searchResult.data.searchResult.forEach((sr) => {
+      sr.items.forEach((eachSCItem) => {
+        const thisSCDetails: StorageClassDetailsType = {
+          name: eachSCItem.name,
+          provisioner: '',
+          storageID: '',
+        };
+        const clusterName = eachSCItem.cluster;
+        const myLabels = (eachSCItem.label?.trim().split(';') || []).map(
+          (lbl): string => {
+            return lbl.trim();
+          }
+        );
+        for (const lbl of myLabels) {
+          const equallySplitLbl = lbl.split('=');
+          if (equallySplitLbl.length === 2) {
+            const [lblKey, lblValue] = equallySplitLbl;
+            if (lblKey === storageIDKey) {
+              thisSCDetails.storageID = lblValue;
+              break;
+            }
+          }
+        }
+        // *************** DELETE BELOW ****************
+        /**/ // console.log("Name of the SC")
+        /**/ // console.log(eachSCItem.name)
+        /**/ eachSCItem.provisioner ??= '';
+        /**/ if (eachSCItem.provisioner === '') {
+          /**/ if (eachSCItem.name.search('cephfs') > -1) {
+            /**/ eachSCItem.provisioner =
+              'openshift-storage.cephfs.csi.ceph.com';
+            /**/
+          } else if (eachSCItem.name.search('ceph-rbd') > -1) {
+            /**/ eachSCItem.provisioner = 'openshift-storage.rbd.csi.ceph.com';
+            /**/
+          }
+          /**/
+        }
+        // *************** DELETE ABOVE ****************
+
+        // filtering StorageClasses with allowed provisioners
+        if (
+          allowedProvisioners.some((allowedProv) => {
+            return eachSCItem.provisioner?.endsWith(allowedProv);
+          })
+        ) {
+          thisSCDetails.provisioner = eachSCItem.provisioner;
+        }
+        // console.log("Modified StorageClass search result")
+        // console.log(eachSCItem)
+        if (
+          thisSCDetails.name !== '' &&
+          thisSCDetails.provisioner !== '' &&
+          thisSCDetails.storageID !== ''
+        ) {
+          const mappedClusterVal =
+            mapClusterNameToSCDetails.get(clusterName) ?? [];
+          mappedClusterVal.push(thisSCDetails);
+          mapClusterNameToSCDetails.set(clusterName, mappedClusterVal);
+        }
+      });
+    });
+  }
+  // console.log("Mapped ClusterName and StorageClass details")
+  // console.log(mapClusterNameToSCDetails)
+  // Make sure we only have TWO clusters
+  // PS: here we are assuming that we can select only TWO clusters
+  if (mapClusterNameToSCDetails.size !== 2) {
+    return [false, []];
+  }
+  // now apply the needed checks
+  // Check A: number of filtered StorageClasses should be same
+  // on both clusters
+  const [cluster1Name, cluster2Name] = [...mapClusterNameToSCDetails.keys()];
```

Comment:
```suggestion
  const [cluster1Name, cluster2Name] = mapClusterNameToSCDetails.keys();
```
mapClusterNameToSCDetails.keys() will return list only.

Reply:
Done.